improver_tests/acceptance/test_checksums.py

```diff
@@ -52,11 +101,38 @@ def test_kgo_checksums():
         filenames = [f for f in filenames if not f.startswith(".")]
         for filename in filenames:
             data_paths.append(pathlib.Path(directory) / filename)
```

Comment:
I know this isn't part of the PR but I think the above code can be made slightly more explicit with the following:

```python
    # generator for list of paths
    data_paths = chain(
        kgo_root.rglob("*.json"),
        kgo_root.rglob("*.json~"),
        kgo_root.rglob("*.nc"),
        kgo_root.rglob("*.nimrod"),
    )
```

Reply:
If you go with this change, need to add `from itertools import chain`.

Reply:
Pathlib is much neater than os.walk!
However, I'd prefer to pick up all files rather than needing a list of specific filename extensions here. I had a go using pathlib glob/rglob, but it looks like the catchall glob `*` behaves different to shell globs and does include dotfiles (eg. `.git`) so won't work for this usage.

Reply:
Yep, that makes sense.